lib/puppet/pops/loader/ruby_legacy_function_instantiator.rb

```diff
@@ -60,4 +63,40 @@ def self.get_binding(loader_injected_arg)
     binding
   end
   private_class_method :get_binding
+
+  def self.assert_code(code_string, result)
+    ripped = Ripper.sexp(code_string)
```

Comment:
The `sexp` method is marked as experimental http://ruby-doc.org/stdlib-2.5.0/libdoc/ripper/rdoc/Ripper.html#method-c-sexp, seems bad to rely on it as a stable API.